neurons/validator.py

```diff
@@ -273,6 +276,61 @@ async def start_dht(self) -> None:
 # API setup
 app = FastAPI(debug=False)
 
+def _get_api_key(request: Request) -> Any:
+    auth_header = request.headers.get("Authorization")
+    if not auth_header:
+        return None
+    if auth_header.startswith("Bearer "):
+        return auth_header.split(" ")[1]
+
+    return auth_header
+
+
+@app.middleware("http")
+async def api_key_validator(request, call_next) -> Response:
+    if request.url.path in ["/docs", "/openapi.json", "/favicon.ico", "/redoc"]:
+        return await call_next(request)
+
+    api_key = _get_api_key(request)
+    if not api_key:
+        return JSONResponse(
+            status_code=HTTP_400_BAD_REQUEST,
+            content={"detail": "API key is missing"},
+        )
+
+    async with db.get_db_connection(db_dir=core_validator.config.db_dir) as conn:
+        api_key_info = await db.get_api_key_info(conn, api_key)
+
+    if api_key_info is None:
+        return JSONResponse(status_code=HTTP_401_UNAUTHORIZED, content={"detail": "Invalid API key"})
+
+    credits_required = 1  # TODO: make this non-constant in the future???? (i.e. dependent on number of pools)????
```

Comment:
Put this in the constants file

Reply:
also the `credits_required` should be dependent on the size of the file. The biggest cost for the validator will be bandwidth